protocol/backend/backend.go

```diff
@@ -28,13 +26,8 @@ import (
 type MessageSender interface {
```

Comment:
I feel like this struct name/comment can be updated

Reply:
The updated name is `NetSender`. It's the interface for someone sending messages via the `net` package's p2p Host, so that name seemed a little more helpful. Feel free to suggest better.